File: TS/TDD-Legacy-Code/E0.Start/Calculator.ts
namespace e0 {

	export enum OperationType {
		Plus,
		Div
	}

	export class Calculator {
		display: string = "";
		lastArgument = 0;
		result = 0;
		newArgument = false;
		shouldReset = true;

		lastOperation: OperationType;

		press(key: string) {
			if (key == "C")
				this.display = "";
			else {
				if (key == "+") {
					this.lastOperation = OperationType.Plus;
					this.lastArgument = Number(this.display);
					this.newArgument = true;
				}
				else {
					if (key == "/") {
						this.lastOperation = OperationType.Div;
						this.lastArgument = Number(this.display);
						this.newArgument = true;
					}
					else if (key == "=") {
						let currentArgument = Number(this.display);
						if (this.lastOperation == OperationType.Plus) {
							this.display = (this.lastArgument + currentArgument).toString();
						}
						if (this.lastOperation == OperationType.Div && currentArgument == 0) {
							this.display = "Division By Zero Error";
						}
						this.shouldReset = true;
					}
					else {
						if (this.shouldReset) {
							this.display = "";
							this.shouldReset = false;
						}
						if (this.newArgument) {
							this.display = "";
							this.newArgument = false;
						}
						this.display += key;
					}
				}
			}
		}

		getDisplay(): string {
			if (this.display == "")
				return "0";
			return this.display;
		}
	}
}

File: Python/e2.fixbug/s3.betterfix/calculator.py
from enum import Enum


class OperationType(Enum):
    Plus = 1
    Div = 2


class Calculator:
    display = ""
    lastArgument = 0
    result = 0
    newArgument = False
    shouldReset = True
    lastOperation = OperationType.Plus

    def press(self, key):
        if key == "C":
            self.display = ""
        else:
            if key == "+":
                if self.display != "":
                    self.lastOperation = OperationType.Plus
                    self.lastArgument = int(self.display)
                    self.newArgument = True
            else:
                if key == "/":
                    if self.display != "":
                        self.lastOperation = OperationType.Div
                        self.lastArgument = int(self.display)
                        self.newArgument = True
                else:
                    if key == "=":
                        if self.display != "":
                            currentArgument = int(self.display)
                            if self.lastOperation == OperationType.Plus:
                                self.display = str(self.lastArgument + currentArgument)
                            if self.lastOperation == OperationType.Div and currentArgument == 0:
                                self.display = "Division By Zero Error"
                            self.shouldReset = True
                    else:
                        if self.shouldReset:
                            self.display = ""
                            self.shouldReset = False
                        if self.newArgument:
                            self.display = ""
                            self.newArgument = False
                        self.display += key

    def get_display(self):
        if self.display == "":
            return "0"
        return self.display
